src/sql/workbench/services/notebook/browser/models/notebookModel.ts

```diff
@@ -538,8 +540,119 @@ export class NotebookModel extends Disposable implements INotebookModel {
 		if (this.inErrorState) {
 			return undefined;
 		}
+
 		let cell = this.createCell(cellType);
+		return this.insertCell(cell, index);
+	}
+
+	public splitCell(cellType: CellType, index?: number, notebookService?: INotebookService): ICellModel | undefined {
+		if (this.inErrorState) {
+			return undefined;
+		}
+
+		let notebookEditor = notebookService.findNotebookEditor(this.notebookUri);
+		let cellEditorProvider = notebookEditor.cellEditors.find(e => e.cellGuid() === this.cells[index].cellGuid);
+		//Only split the cell if the markdown editor is open.
+		//TODO: Need to handle splitting of cell if the selection is on webview
+		if (cellEditorProvider) {
+			let editor = cellEditorProvider.getEditor() as QueryTextEditor;
+			if (editor) {
+				let editorControl = editor.getControl() as CodeEditorWidget;
+
+				let model = editorControl.getModel() as TextModel;
+				let range = model.getFullModelRange();
+				let selection = editorControl.getSelection();
+				let source = this.cells[index].source;
+				let newcell = null, tailcell = null, partial = null;
+				let newcellindex = index;
+				let tailcellindex = index;
+
+				// Save UI state
+				let showMarkdown = this.cells[index].showMarkdown;
+				let showPreview = this.cells[index].showPreview;
+
+				//Get selection value from current cell
+				let newcellcontent = model.getValueInRange(selection);
+
+				//Get content after selection
+				let tailrange = range.setStartPosition(selection.endLineNumber, selection.endColumn);
+				let tailcellcontent = model.getValueInRange(tailrange);
+
+				//Get content before selection
+				let headrange = range.setEndPosition(selection.startLineNumber, selection.startColumn);
+				let headcontent = model.getValueInRange(headrange);
+
+				//Set content before selection if the selection is not the same as original content
```

Comment:
I don't see any logic that is doing anything if the selection is the same as the original content

Reply:
Good catch

Reply:
I'm still not seeing any special logic around that here. Did you mean to remove that comment or something?

Reply:
Check Line #586 - #588 which was added

Reply:
Oh...ok. I don't think this comment is necessary then (since the one on 585 already tells the user what's happening) but I'll leave that up to you.